### PR TITLE
MINOR: Remove TLS renegotiation code

### PR DESCRIPTION
**Pull request by ijuma**
This has been disabled since the start and since
it's removed in TLS 1.3, there are no plans to
ever support it.

### CODE REVIEW

**Comment by ijuma** [comment]
Thoughts @rajinisivaram?

File: clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java [hunk 1]
```diff
@@ -240,9 +237,8 @@ protected boolean flush(ByteBuffer buf) throws IOException {
     */
     @Override
     public void handshake() throws IOException {
-        // Reset state to support renegotiation. This can be removed if renegotiation support is removed.
-        if (state == State.READY)
-            state = State.HANDSHAKE;
+        if (state != State.HANDSHAKE)
+            throw renegotiationException();
```
Comment (rajinisivaram) [inline clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java]: We can also get here if handshake has already failed (state == `State.HANDSHAKE_FAILED`) and there are still bytes to be flushed in `netWriteBuffer`.

**Review by rajinisivaram — COMMENTED** [review]
@ijuma Thanks for the PR. There are a couple of unused imports in `SslSelectorTest` causing the builds to fail. Also left a comment in `handshake()`.

File: clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java [hunk 2]
```diff
@@ -240,9 +237,10 @@ protected boolean flush(ByteBuffer buf) throws IOException {
     */
     @Override
     public void handshake() throws IOException {
-        // Reset state to support renegotiation. This can be removed if renegotiation support is removed.
         if (state == State.READY)
-            state = State.HANDSHAKE;
+            throw renegotiationException();
+        if (state == State.CLOSING)
+            throw closingException();
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java:243]: This throws a different exception. Should we throw the same one?

Comment (rajinisivaram) [reply clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java:243]: I think at the moment, we should never get here, so `IllegalStateException` is fine.

**Review by rajinisivaram — APPROVED** [review]
LGTM

**Comment by ijuma** [comment]
Merging to trunk.

**Pull request closed without merging**